### DISCUSSION TITLE
Doc.similarity either outputting 0 or 1 (only when it's the same string)

### DISCUSSION THREAD

**Discussion by JayThibs** [explosion/spaCy#11116] [Help: Coding & Implementations] [answered]
## How to reproduce the behaviour

When I run the code below, I get a score of 0.0. I only get a score of 1 if I use "apple" for doc2. Everything else it 0.

```
!pip install -U pip setuptools wheel --quiet
!pip install -U 'spacy[cuda113]' --quiet
!python -m spacy download en_core_web_trf --quiet
import spacy
nlp = spacy.load("en_core_web_trf")
doc1 = nlp(u"apple")
doc2 = nlp(u"green apple")
doc1.similarity(doc2)
```

Any ideas what's wrong?

## Your Environment
* Operating System: Colab
* Python Version Used: 3.7
* spaCy Version Used: Latest

**Comment by polm** [comment, marked as answer]
This is expected behavior. `doc.similarity` relies on `doc.vector`, which relies on word vectors, and the Transformers pipeline has no word vectors. See #9011.
